test/test_batterystate.py

```diff
@@ -0,0 +1,282 @@
+        msg = BatteryState()
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.FAIL)
+
+    def test_charging(self):
+        msg = BatteryState()
+        # Charging at max, report ACTIVE
+        msg.power_supply_status = BatteryState.POWER_SUPPLY_STATUS_CHARGING
+        msg.percentage = 1.
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.ACTIVE)
+
+        # Charging but above max, report SUCCESS
+        self.blackboard.save("max_charge", 0.95, self.charge_monitor._id)
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.SUCCESS)
+
+        # Charging but below max, report ACTIVE
+        msg.percentage = 0.
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)  # let callbacks finish
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.ACTIVE)
+
+    def test_discharging(self):
+        msg = BatteryState()
+        # Discharging at max, report FAIL
+        msg.power_supply_status = BatteryState.POWER_SUPPLY_STATUS_DISCHARGING
+        msg.percentage = 1.
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.FAIL)
+
+        # Discharging but above max, report FAIL
+        self.blackboard.save("max_charge", 0.95, self.charge_monitor._id)
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.FAIL)
+
+        # Discharging but below max, report FAIL
+        msg.percentage = 0.
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)  # let callbacks finish
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.FAIL)
+
+    def test_full(self):
+        msg = BatteryState()
+        # Charge at full, report SUCCESS.
+        msg.power_supply_status = BatteryState.POWER_SUPPLY_STATUS_FULL
+        msg.percentage = 1.
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.SUCCESS)
+
+        # Charge at full but above max, report SUCCESS
+        self.blackboard.save("max_charge", 0.95, self.charge_monitor._id)
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.SUCCESS)
+
+        # Charge at full, but percentage low, still report success
+        msg.percentage = 0.
+        self.msg_pub.publish(msg)
+        rospy.sleep(0.1)  # let callbacks finish
+        result = self.charge_monitor.tick()
+        assert_equal(result.status, NodeStatus.SUCCESS)
```

Comment:
this test is funny to me, when would be see a full status with a low percentage?